lib/OpenQA/BuildResults.pm

```diff
@@ -109,34 +107,57 @@ sub compute_build_results {
         $group_ids = [$group->id];
     }
 
+    my %builds;
+    my %result = (
+        result   => \%builds,
+        max_jobs => 0,
+        children => [map { {id => $_->id, name => $_->name} } @children],
+        group    => {
+            id   => $group->id,
+            name => $group->name
+        });
+
+    if (defined($limit) && int($limit) <= 0) {
+        return \%result;
```

Comment:
It is not possible to pass `rows => 0` to the database. But if limit is 0 the database query can be skipped anyways and that's what this is for.